src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/RazorDynamicFileInfoProvider.cs

```diff
@@ -67,26 +68,46 @@ public RazorDynamicFileInfoProvider(Lazy<RazorWorkspaceListenerInitializer> razo
         Contract.ThrowIfNull(LanguageServerHost.Instance, "We don't have an LSP channel yet to send this request through.");
         var clientLanguageServerManager = LanguageServerHost.Instance.GetRequiredLspService<IClientLanguageServerManager>();
 
-        var response = await clientLanguageServerManager.SendRequestAsync<ProvideDynamicFileParams, ProvideDynamicFileResponse>(
+        var response = await clientLanguageServerManager.SendRequestAsync<RazorProvideDynamicFileParams, RazorProvideDynamicFileResponse>(
             ProvideRazorDynamicFileInfoMethodName, requestParams, cancellationToken);
 
-        // Since we only sent one file over, we should get either zero or one URI back
-        var responseUri = response.CSharpDocument?.Uri;
-
-        if (responseUri == null)
+        if (response.CSharpDocument is null)
         {
             return null;
         }
-        else
+
+        // Since we only sent one file over, we should get either zero or one URI back
+        var responseUri = response.CSharpDocument.Uri;
+        var dynamicFileInfoFilePath = ProtocolConversions.GetDocumentFilePathFromUri(responseUri);
+
+        if (response.Edits is not null)
         {
-            var dynamicFileInfoFilePath = ProtocolConversions.GetDocumentFilePathFromUri(responseUri);
-            return new DynamicFileInfo(dynamicFileInfoFilePath, SourceCodeKind.Regular, EmptyStringTextLoader.Instance, designTimeOnly: true, documentServiceProvider: null);
+            var project = _workspaceFactory.Workspace.CurrentSolution.GetRequiredProject(projectId);
+            var document = project.Documents.FirstOrDefault(
+                d => d.FilePath is not null && PathUtilities.PathsEqual(d.FilePath, dynamicFileInfoFilePath));
```

Comment:
you should just be able to use `https://github.com/dotnet/roslyn/blob/main/src/LanguageServer/Protocol/Extensions/Extensions.cs#L103` on the solution